src/App.tsx

```diff
@@ -1,222 +1,183 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import {
+  getTodos,
+  updateTodo,
+  createTodo,
+  deleteTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo, TodoTitle } from './types/Todo';
+import { ErrorType } from './components/ErrorType/ErrorType';
+
+enum SelectedType {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>(SelectedType.ALL);
+  const [showAll, setShowAll] = useState(true);
+  const [isError, setIsError] = useState('');
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const handleError = (text: string) => {
+    setIsError(text);
+    setTimeout(() => {
+      setIsError('');
+    }, 5000);
+  };
+
+  const handleNewTodos = async () => {
+    if (query.trim().length === 0) {
+      handleError('Type something');
+
+      return;
+    }
+
+    try {
+      const newTodo = await createTodo(user?.id, query);
+
+      setTodos((prevState) => {
+        return [...prevState, newTodo];
+      });
+    } catch (error) {
+      handleError('Can not add a todo');
+    }
+  };
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      await deleteTodo(todoId);
+      setTodos(prevState => prevState.filter((a) => a.id !== todoId));
+    } catch (error) {
+      handleError('Can not delete a todo');
+    }
+  }, []);
+
+  const changeTodo = useCallback(async (todoId: number, object: TodoTitle) => {
+    try {
+      const updatedTodo: Todo = await updateTodo(todoId, object);
+
+      setTodos(prevState => (prevState.map((a) => (a.id === todoId
+        ? updatedTodo
+        : a))
+      ));
+    } catch (error) {
+      handleError('Can not update a todo');
+    }
+  }, []);
+
+  const showAllTodos = () => {
+    todos.forEach(todo => {
+      changeTodo(todo.id, { completed: showAll });
+    });
+
+    setShowAll(!showAll);
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    getTodos(user?.id)
+      .then(res => setTodos(res))
+      .catch(() => isError);
```

Comment:
Do you understand what this catch will do? Maybe you need to use `setIsError` here instead?